app/main/controllers/activity/RTMediaActivity.php

```diff
@@ -53,71 +68,110 @@ function create_activity_html( $type = 'activity' ) {
-			if ( 'photo' === $media->media_type || 'document' === $media->media_type || 'other' === $media->media_type  ) {
-				$li_content .= '<a href ="' . esc_url( get_rtmedia_permalink( $media->id ) ) . '">';
-			}
-			$li_content .= '<div class="rtmedia-item-thumbnail">';
+		$rtmedia_activity_ul_class = apply_filters( 'rtmedia_' . $type . '_ul_class', 'rtm-activity-media-list' );
 
-			$li_content .= $this->media( $media );
-
-			$li_content .= '</div>';
-
-			$li_content .= '<div class="rtmedia-item-title">';
-			$li_content .= '<h4 title="' . esc_attr( $media->media_title ) . '">';
-			if ( 'photo' !== $media->media_type && 'document' !== $media->media_type && 'other' !== $media->media_type ) {
-				$li_content .= '<a href="' . esc_url( get_rtmedia_permalink( $media->id ) ) . '">';
-			}
+		$media_content = '';
+		$count         = 0;
+		foreach ( $media_details as $media ) {
+			$media_content .= sprintf( '<li class="rtmedia-list-item media-type-$s">', esc_attr( $media->media_type ) );
 
-			$li_content .= $media->media_title;
-			if ( 'photo' !== $media->media_type && 'document' !== $media->media_type && 'other' !== $media->media_type ) {
-				$li_content .= '</a>';
-			}
-			$li_content .= '</h4>';
-			$li_content .= '</div>';
-			if ( 'photo' === $media->media_type || 'document' === $media->media_type || 'other' === $media->media_type ) {
-				$li_content .= '</a>';
+			if ( 'photo' === $media->media_type ) {
+				// Markup for photo media type with anchor tag only on image.
+				$media_content .= sprintf(
+					'<a href ="%1$s">
+						<div class="rtmedia-item-thumbnail">
+							%2$s
+						</div>
+						<div class="rtmedia-item-title">
+							<h4 title="%3$s">
+								%4$s
+							</h4>
+						</div>
+					</a>',
+					esc_url( get_rtmedia_permalink( $media->id ) ),
+					$this->media( $media ),
+					esc_attr( $media->media_title ),
+					$media->media_title
+				);
+			} elseif ( 'audio' === $media->media_type || 'video' === $media->media_type ) {
+				// Markup for audio and video media type with link only on media (title).
+				$media_content .= sprintf(
+					'<div class="rtmedia-item-thumbnail">
+						%1$s
+					</div>
+					<div class="rtmedia-item-title">
+						<h4 title="%2$s">
+							<a href="%3$s">
+								%4$s
+							</a>
+						</h4>
+					</div>',
+					$this->media( $media ),
+					esc_attr( $media->media_title ),
+					esc_url( get_rtmedia_permalink( $media->id ) ),
+					$media->media_title
+				);
+			} else {
+				// Markup for all the other media linke docs and other files where anchor tag the markup is comming from add-on itself.
+				$media_content .= sprintf(
+					'<div class="rtmedia-item-thumbnail">
+							%1$s
+					</div>
+					<div class="rtmedia-item-title">
+							<h4 title="%2$s">
+								%3$s
+							</h4>
+					</div>',
+					$this->media( $media ),
+					esc_attr( $media->media_title ),
+					$media->media_title
+				);
```

Comment:
More optimizable, but less customizable way.

```
$media_title = ( 'audio' === $media->media_type || 'video' === $media->media_type ) ? sprintf( '<a href="%s">%s</a>', esc_attr( $media->media_title ) ) : $media->media_title;

$media_inner_content =  sprintf(
                    '<div class="rtmedia-item-thumbnail">
                            %1$s
                    </div>
                    <div class="rtmedia-item-title">
                            <h4 title="%2$s">
                                %3$s
                            </h4>
                    </div>',
                    $this->media( $media ),
                    esc_attr( $media->media_title ),
                    $media_title
                );
if ( 'photo' === $media->media_type ) {
      $media_inner_content = sprintf( '<a href ="%1$s">%2$s</a>', $media_inner_content );
}

$media_content .= $media_inner_content
```

Reply:
I'm keeping it as it is, so it'll be easy to add modification in the future.

Reply:
Try to reuse the code here if possible, as if it's using same markup then change would be easy ( i.e at one place ), if a different type needs any specific change then we can implement as you have done now.